common/workunit/workunit.cpp

```diff
@@ -1023,6 +1023,151 @@ class CConstGraphProgressScopeIterator : public CInterfaceOf<IConstWUScopeIterat
 };
 
 
+int compareNoteScopeOrder(IConstWUException & left, IConstWUException & right)
+{
+    return compareScopeName(left.queryScope(), right.queryScope());
+}
+
+int compareNoteScopeOrder(IInterface * const * left, IInterface * const * right)
+{
+    return compareNoteScopeOrder(*static_cast<IConstWUException *>(*left), *static_cast<IConstWUException *>(*right));
+}
+
+class NotesIterator : public CInterfaceOf<IConstWUScopeIterator>
```

Comment:
At some point some of this code could be commoned up with the global stats - it it was thought worthwhile.  Probably when the next iterator comes along.

Reply:
Noted